src/chat.rs

```diff
@@ -1482,7 +1482,10 @@ pub unsafe fn remove_contact_from_chat(
         "bad chat_id = {} <= 9",
         chat_id
     );
-    ensure!(contact_id != DC_CONTACT_ID_SELF, "Cannot remove self");
+    ensure!(
+        !(contact_id <= DC_CONTACT_ID_LAST_SPECIAL && contact_id != DC_CONTACT_ID_SELF),
```

Comment:
this is hard to read, better to split this into `contact_id > SPECIAL || contact_id  == self`